metrix/src/parser.rs

```diff
@@ -1,11 +1,14 @@
-use nom;
-
 use std::str;
 use std::fmt;
 
 use nom::{
+    branch::alt,
+    bytes::complete::{take_while1, tag},
```

Comment:
ah fuck, this is greek to me. i need to read up on nom

Reply:
lol good luck. It's actually a pretty sick lib, but their docs do leave you a little wanting. Give nthat it's such an incredible widely applicable library, you'd hope for more specific use case documentation, but alas, only 3 in-repo examples with limited usage coverage

Reply:
The short story of what I'm doing here is dipping into the actual nom 5 functionality and structure. It's all function based, which actually looks a lot cleaner, and is a little more flexible. Extra bonus is it's actually notably faster as well.